lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -159,16 +154,45 @@ def setup(self):
 
         self.log.debug(f"Output DL3 file: {self.output_file}")
 
+        try:
+            QTable.read(self.input_irf, hdu=1).meta["GH_CUT"]
```

Comment:
There is no nead to read the full table to just look at the fits header.
Since we also need at least one of the two, it's enough here to check the header keyword.

So just use:

```python
from astropy.io import fits

with fits.open(self.input_irf) as hdul:
    self.energy_dependent_gh_cut = "GH_CUT" not in hdul["EFFECTIVE AREA"].header
```

To get the information whether the IRFs are pointlike or not, you can look at:

```

with fits.open(self.input_irf) as hdul:
    aeff_hdu = hdul['EFFECTIVE_AREA']
    self.energy_dependent_gh_cut = 'GH_CUT' not in aeff_hdu.header
    self.is_pointlike = aeff_hdu.header['HDUCLAS3'] == 'POINT-LIKE'
    if self.is_pointlike:
        self.energy_dependent_theta_cut = 'RAD_MAX' not in aeff_hdu.header
```

Reply:
Sorry, this is the create dl3 file, so no theta cut needed.

Reply:
Thanks! Will amend this